server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -113,38 +100,28 @@ public RemoteStoreRefreshListener(
         this.remoteDirectory = (RemoteSegmentStoreDirectory) ((FilterDirectory) ((FilterDirectory) indexShard.remoteStore().directory())
             .getDelegate()).getDelegate();
         localSegmentChecksumMap = new HashMap<>();
-        RemoteSegmentMetadata remoteSegmentMetadata = null;
-        if (indexShard.routingEntry().primary()) {
-            try {
-                remoteSegmentMetadata = this.remoteDirectory.init();
```

Comment:
I assume we are removing `init` as it is already triggered in `RemoteSegmentTrackerListener`. Is that correct? 
In that case, these 2 listeners are very tightly coupled. We need to make sure one does not break if we are making changes to other.

Reply:
> I assume we are removing init as it is already triggered in RemoteSegmentTrackerListener. Is that correct?

Yes, the init method has been moved.

> In that case, these 2 listeners are very tightly coupled. We need to make sure one does not break if we are making changes to other.

This is one time operation during the ctor invocation. Also, I am evaluating if the init method is really required given we are running init method in ctor of RemoteSegmentStoreDirectory itself.

Reply:
Also, can there be race condition here? how are we sure that `RemoteSegmentTrackerListener` ctor will execute before ctor of `RemoteStoreRefreshListener`.

+1 on the tight coupling. Since the goal of the PR is the separate out Local and Remote tracker updates, we have dependency on RemoteSegmentTrackerListener for `getLatestLocalFileNameLengthMap` and `getPrimaryTermAtInit`

Reply:
> Also, can there be race condition here? how are we sure that RemoteSegmentTrackerListener ctor will execute before ctor of RemoteStoreRefreshListener.

The order of constructor is maintained. 

> +1 on the tight coupling. Since the goal of the PR is the separate out Local and Remote tracker updates, we have dependency on RemoteSegmentTrackerListener for getLatestLocalFileNameLengthMap and getPrimaryTermAtInit

We have multiple ITs and UTs in place to detect issues if there can be potential issue. Objects being shared does not seem to be a valid reason for coupling.